packages/js/src/generators/release-version/release-version.ts

```diff
@@ -27,12 +28,13 @@ import * as ora from 'ora';
 import { relative } from 'path';
 import { prerelease } from 'semver';
 import { ReleaseVersionGeneratorSchema } from './schema';
+import { packageManagerInstall } from './utils/package-manager-install';
 import { resolveLocalPackageDependencies } from './utils/resolve-local-package-dependencies';
 
 export async function releaseVersionGenerator(
   tree: Tree,
   options: ReleaseVersionGeneratorSchema
-) {
+): Promise<ReleaseVersionGeneratorResult> {
```

Comment:
In a dedicated follow up (because it needs some consideration around trying not to break existing usage), we can maybe generalize an updated `GeneratorResult` interface.

Currently a generator can return `void | GeneratorCallback` I believe.

I think it's reasonable that any generator could return arbitrary data, as we've seen already with versioning, and as we know we still need the `GeneratorCallback`, so maybe in the future:

```ts
generator(...): void | GeneratorCallback | GeneratorResult;

interface GeneratorResult {
  callback?: GeneratorCallback;
  data?: Record<string, unknown>;
}
```

And maybe even in v18 just `generator(...): GeneratorResult;` with the two optional properties

Reply:
We can do this in a follow up. The existing GeneratorCallback will need to be changed because it currently doesn't accept any arguments and always returns `void` or `Promise<void>`, whereas this callback needs to return data.

Reply:
Something else to note is that in the case of `nx release`, the version generator is called once per release group. We don't actually want to perform the lock file update once per release group though, just once at the end of all updates. The lock file update callback can be a static property on the generator.